### PR TITLE
feat: Initial compatibility and tests for COP23

### PR DESCRIPTION
**Pull request by jason-p-pickering**
### Summary of Proposed Changes

- Upgrades CI testing environment to R 4.2.1
- Added function `writeSpectrumData` which can be used to populate the DataPack Spectrum tab during testing. 
- Fixed bug in `unPackSchema`related to detecting invalid column and value types.
- Fixed create schema unit test.
- Removed superfluous warning related to missing PSNUs. 
- Fixed issue in `unPackingChecks` related to the lack of the SNU1 column in COP23 tools.
- Fixed a testing issue related to choosing the correct template to use for testing. 
- Added testing helper functions to deal with peculiarities of the CI testing environment. 
- Added various tests and  testing files for COP23 tools.
- Updated several unit tests to favor COP23 over COP21.
- Disabled two unit tests for COP21. 
- Alters test method from `devtools::check` to `devtools::test`, which skips CRAN package checks. 
- Adds initial parsing methods for COP23 Year2 tabs

### Related Issues
- DP-XXX
- etc.

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

File: R/unPackingChecks.R [hunk 1]
```diff
@@ -780,7 +780,7 @@ checkInvalidOrgUnits <- function(sheets, d, quiet = TRUE) {
 
   invalid_orgunits <- d$sheets[sheets] %>%
     dplyr::bind_rows(.id = "sheet_name") %>%
-    dplyr::filter(dplyr::if_any(c("SNU1", "PSNU", "Age", "Sex"), ~!is.na(.))) %>%
+    dplyr::filter(dplyr::if_any(c("PSNU", "Age", "Sex"), ~ !is.na(.))) %>%
```
Comment (jason-p-pickering) [inline R/unPackingChecks.R:783]: This SNU1 seems to be missing...not sure why its there to begin with?

Comment (jason-p-pickering) [reply R/unPackingChecks.R]: Added some conditional logic based on the COP year to deal with this.

**Comment by jason-p-pickering** [comment]
@JordanBalesBAO @flopez-bao I have discussed with @christian-domaas and we would like to go head and merge this into dev and then make an initial release. Can you guys review the PR and provide your comments / approval?

**Comment by JordanBalesBAO** [comment]
@jason-p-pickering added this to my list.

**Review by JordanBalesBAO — APPROVED** [review]

**Comment by JordanBalesBAO** [comment]
Rather dense, but all looks good to me.

**Pull request merged**